feat(dashboard): created UI for viewing courses on the dashboard

mitchfaulcon opened the pull request:
### Purpose

<!---
Describe the problem or feature.
Consider providing an overview of why the work is taking place (with any relevant links); don’t assume familiarity with the history.
-->
Implemented the frontend for the dashboard page for viewing courses

<!---
Link to the related issues [KEY-000](link)
e.g [HRT-27](https://softeng761team5.atlassian.net/browse/HRT-27)
--->

[HRT-64](https://softeng761team5.atlassian.net/browse/HRT-64)

<!-- []() -->
<!-- []() -->

### Approach

<!---
How does this change address the problem
--->
- Uses the dummy course data to populate a flexbox of course cards
- These can be clicked on to be redirected after the backend is hooked up
- Added a create course button that is always at the end of the course cards in the flexbox
- **Note: There is a bug where the cards don't stay centered when shrinking the window horizontally. I'm not sure how to fix it but if anyone wants to have a look and let me know that'd be great :)**

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!--
If this PR contains a breaking change, describe the impact and migration path for existing applications below.
-->

### Screenshots before and after this change (required for UI changes)

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

<!---
Required if there is UI change
--->
![image](https://user-images.githubusercontent.com/42291691/94647919-fc963680-034d-11eb-997b-a717fbaa3626.png)

#### Screenshot(s) of UI after this PR

<!---
Required if there is UI change
--->
![image](https://user-images.githubusercontent.com/42291691/94647840-d6709680-034d-11eb-943d-1388a969bb72.png)
![image](https://user-images.githubusercontent.com/42291691/94647876-e7210c80-034d-11eb-8b74-fb717b6edc8d.png)

mitchfaulcon committed:
feat(dashboard): created UI for viewing courses on the dashboard

mitchfaulcon committed:
Merge remote-tracking branch 'origin/master' into HRT-64-create-dashboard-UI

mitchfaulcon committed:
refactor(dashboard.tsx): removed unused import

gargi-bansal commented on src/components/Course.tsx:
```diff
@@ -0,0 +1,23 @@
+
+const Dimension: React.FC<CourseProps> = (props: CourseProps) => {
+  const onClick = () => {
+    // TODO: Redirect to appropriate course page
```
Is this going to be a different ticket (the course page on the dashboard)? It's not currently on the Jira backlog

mitchfaulcon replied:
Yeah I think it is, I'll add a new ticket for it

gargi-bansal commented on src/pages/Dashboard.tsx:
```diff
@@ -23,15 +27,38 @@ const Dashboard: React.FC<RouteComponentProps> = (props) => {
+          className="Instructions-Button"
+          onClick={onInstructionsClick}
+        >
+          How to play
```
I think this button now needs a shadow since it is on the main background

gargi-bansal reviewed (changes requested):
Dashboard looks great! Just one small request. Can the courses be centered, it looks weird when I minimise the window a bit.
<img width="1287" alt="Screen Shot 2020-10-01 at 11 43 14 AM" src="https://user-images.githubusercontent.com/42624632/94747437-614ba280-03db-11eb-99a6-740be4fa7a44.png">

mitchfaulcon committed:
feat(dashboard.css): added shadow to instructions button

gargi-bansal committed:
feat(dashboard): centered courses on dashboard

fixed styling to center courses on dashboard

gargi-bansal committed:
feat(dashboard): fixed centering of courses on dashboard page

fixed centering of courses so that they center while still sticking to the left on incomplete rows

gargi-bansal committed:
Merge branch 'HRT-64-create-dashboard-UI' of https://github.com/HeartUoA/HeaRT into HRT-64-create-dashboard-UI

mitchfaulcon committed:
fix(dashboard): fixed cards not centering on resize

preetpatel commented on src/pages/Dashboard.tsx:
```diff
@@ -1,37 +1,84 @@
+        >
+          {courses.map((item) => {
+            return (
+              <Course
```
right here.. you need to give that course an ID.. so you can just pull it from `item._id`

mitchfaulcon replied:
using the key property?

preetpatel replied:
yup

gargi-bansal replied:
yup

mitchfaulcon replied:
thanks :)

mitchfaulcon committed:
refactor(dashboard): moved create course card to start of cards

mitchfaulcon committed:
fix(dashboard): added course id as key property for cards

preetpatel committed:
Merge branch 'master' into HRT-64-create-dashboard-UI

mitchfaulcon committed:
feat(course.css dashboard.css): change card colour on hover
